hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFileReader.java

```diff
@@ -259,6 +254,11 @@ private HoodieLogBlock readBlock() throws IOException {
               contentPosition, contentLength, blockEndPos, readerSchema, header, footer, keyField);
         }
       case HFILE_DATA_BLOCK:
+        if (HoodieTableMetadata.isMetadataTable(logFile) && keyDeDuplication) {
+          return new HoodieMetadataHFileDataBlock(logFile, inputStream, Option.ofNullable(content), readBlockLazily,
```

Comment:
Right, we want to avoid any on-disk format/type changes like new block type code that gets persisted in the log block. 

```
  public enum HoodieLogBlockType {
    COMMAND_BLOCK, DELETE_BLOCK, CORRUPT_BLOCK, AVRO_DATA_BLOCK, HFILE_DATA_BLOCK
  }
```

This PR uses the existing HFILE_DATA_BLOCK type only. The new classes in the PR are only code org to make the metadata HFile reading and writing override the base HFile reader/block for the needed functionality.

Reply:
Not sure if we explicitly discussed this, but I think this can be done as just the reader config. i.e this layer should not be aware of whether its a MT or data table.